commcare_connect/opportunity/payment_number_report.py

```diff
@@ -0,0 +1,108 @@
+from collections import defaultdict
+
+import django_filters
+import django_tables2 as tables
+from django.http import HttpResponse
+from django.urls import reverse
+from django.utils.html import format_html
+
+from commcare_connect.connect_id_client.main import fetch_payment_phone_numbers, validate_phone_numbers
+from commcare_connect.opportunity.models import Opportunity, OpportunityAccess
+from commcare_connect.reports.views import NonModelTableBaseView, SuperUserRequiredMixin
+
+
+class PaymentNumberReportTable(tables.Table):
+    username = tables.Column(verbose_name="Username")
+    phone_number = tables.Column(verbose_name="Payment Phone Number")
+    status = tables.Column(verbose_name="Status")
+
+    class Meta:
+        template_name = "opportunity/payment_numbers_table.html"
+        attrs = {"class": "table table-striped table-hover"}
+        empty_text = "No data available."
+        orderable = False
+
+    def render_status(self, value, record):
+        options = ["pending", "approved", "rejected"]
+        username = record["username"]
+        phone_number = record["phone_number"]
+
+        radio_buttons = [
+            f'<input type="radio" name="status_{username}" value="{option}" {"checked" if value == option else ""}> {option.capitalize()}'
+            for option in options
+        ]
+        radio_buttons.append(f'<input type="hidden" name="phone_{username}" value="{phone_number}">')
+        return format_html("<br>".join(radio_buttons))
+
+
+class PaymentFilters(django_filters.FilterSet):
+    STATUS_CHOICES = [
+        ("pending", "Pending"),
+        ("approved", "Approved"),
+        ("rejected", "Rejected"),
+    ]
+
+    opportunity = django_filters.ChoiceFilter(method="filter_by_ignore")
+    status = django_filters.ChoiceFilter(choices=STATUS_CHOICES, label="Payment Status")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        opportunities = Opportunity.objects.filter(
+            organization=self.request.org, payment_info_required=True
+        ).values_list("id", "name")
+        self.filters["opportunity"] = django_filters.ChoiceFilter(
+            choices=opportunities, label="Opportunity", empty_label=None
+        )
+
+    class Meta:
+        model = None
+        fields = ["status"]
+
+    def filter_by_ignore(self, queryset, name, value):
+        return queryset
+
+
+class PaymentNumberReport(tables.SingleTableMixin, SuperUserRequiredMixin, NonModelTableBaseView):
```

Comment:
I was planning to update the role here and forgot. I will change the permissions to org-user (can manage opportunities)

Reply:
Can you explain a little more about the expected workflows and security model? It seems like if a user is in a multiple different opportunities across orgs, any member of any of those orgs can impact their ability to get paid by the other org by changing this value.

Reply:
My concern is that they are setting a global property for the user that affects other opportunities as well. I think edit opportunity/org member permissions are reasonable for changes that are isolated to that opportunity, but here the actions they take affect the other opportunities the user is a part of, and their ability to deliver or be paid in those opportunities as well.

Reply:
True. Payment profiles are not org/opportunity specific, but any org with payment-info-required opportunities will be able to manage them.

I believe we discussed this earlier in the weekly calls (during the spec process) and we were fine with an org being able to say that a payment number can or can not receive payments. (Update: I have double checked with Rama again on this.)

This tool makes it necessary to select an opportunity (only from those that have payments enabled) and then fetches users' payment profiles for that opportunity [only](https://github.com/dimagi/commcare-connect/pull/428/files#diff-402f3f089eec80ce6c41172291a5f798efc9806cf95f6ea8a07556420d1bb708R85) for managing.

Reply:
> we were fine with an org being able to say that a payment number can or can not receive payments.

I recall being fine with orgs blocking payment _for their opportunity_, I am very concerned about their ability to block it for _other_ opportunities, especially ones they are active in, and potentially already getting paid. Happy to take this offline, but I do not believe it is reasonable behavior to have different orgs be able to override each other's work here, and I think this approval should probably be org or opportunity specific. That would also address the concern previously raised about these numbers no longer working over time, since they would be approved each time.